packages/ui/src/components/theme-selector-v2/index.tsx

```diff
@@ -0,0 +1,229 @@
+import { useEffect, useState } from 'react'
+
+import { Button, ButtonGroup, Dialog, Separator, Text } from '@/components'
+import { cn } from '@/utils/cn'
+
+export interface ThemeInterface {
+  mode: Mode
+  contrast: Contrast
+  colorAdjustment: ColorAdjustment
+  accentColor: AccentColor
+}
+
+export enum Mode {
+  Dark = 'Dark',
+  Light = 'Light'
+}
+
+export enum Contrast {
+  Default = 'Default',
+  HighContrast = 'High Contrast',
+  Dimmer = 'Dimmer'
+}
+
+export enum ColorAdjustment {
+  Default = 'Default',
+  Protanopia = 'Protanopia',
+  Deuteranopia = 'Deuteranopia',
+  Tritanopia = 'Tritanopia'
+}
+
+export enum AccentColor {
+  Yellow = 'yellow',
+  Red = 'red',
+  Pink = 'pink',
+  Purple = 'purple',
+  Blue = 'blue',
+  Cyan = 'cyan',
+  Orange = 'orange',
+  Brown = 'brown',
+  Green = 'green'
+}
+
+interface ThemeDialogProps {
+  defaultTheme?: ThemeInterface
+  theme?: ThemeInterface
+  open: boolean
+  onOpenChange: (open: boolean) => void
+  onChange: (language: Partial<ThemeInterface>) => void
+  onSave: (language: ThemeInterface) => void
+  onCancel: () => void
+  children: React.ReactNode
+}
+
+const ThemeDialog: React.FC<ThemeDialogProps> = ({
+  defaultTheme,
+  theme,
+  open,
+  onOpenChange,
+  onChange,
+  onSave,
+  onCancel,
+  children
+}) => {
+  const [mode, setMode] = useState<Mode>(Mode.Dark)
+  const [contrast, setContrast] = useState<Contrast>(Contrast.Default)
+  const [colorAdjustment, setColorAdjustment] = useState<ColorAdjustment>(ColorAdjustment.Default)
+  const [accentColor, setAccentColor] = useState<AccentColor>(AccentColor.Blue)
+
+  useEffect(() => {
+    if (theme) {
+      setMode(theme.mode)
+      setContrast(theme.contrast)
+      setColorAdjustment(theme.colorAdjustment)
+      setAccentColor(theme.accentColor)
+    } else if (defaultTheme) {
+      setMode(defaultTheme.mode)
+      setContrast(defaultTheme.contrast)
+      setColorAdjustment(defaultTheme.colorAdjustment)
+      setAccentColor(defaultTheme.accentColor)
+    }
+  }, [defaultTheme, theme])
+
+  return (
+    <Dialog.Root open={open} onOpenChange={onOpenChange}>
+      <Dialog.Trigger asChild>{children}</Dialog.Trigger>
```

Comment:
Instead of controlling the open state externally and internally, it's probably best to allow the `Dialog` to handle its own open/closed state.

Reply:
if possible, we could mirror radix, which supports both.  you can add a `<Dialog.Trigger>` and it will automatically handle open / close state.  Or you can pass in `open={true}` and handle it externally.  But I don't want to over complicate ...

Reply:
@bradrydzewski current solution supports both, if `open` prop is specified, it's controlled and Dialog opens/closes based on the value of this prop. If `prop` is not specified, the trigger `children` would click the Dialog.